server.js

```diff
@@ -3770,12 +3852,15 @@ function sendMessagetoEditors(msgContent, sourceSocketID) {
     // console.log(Object.keys(realityEditorSocketArray).length + ' editor sockets connected');
 
     for (var thisEditor in realityEditorSocketArray) {
-        if (typeof sourceSocketID !== 'undefined' && thisEditor === sourceSocketID) {
-            continue; // don't trigger the read listener of the socket that originally wrote the data
-        }
-        if (msgContent.object === realityEditorSocketArray[thisEditor].object && msgContent.frame === realityEditorSocketArray[thisEditor].frame) {
-            messagetoSend(msgContent, thisEditor);
-        }
+        realityEditorSocketArray[thisEditor].forEach((thisObj) => {
+            if (typeof sourceSocketID !== 'undefined' && thisEditor === sourceSocketID && msgContent.object === thisObj.object && msgContent.frame === thisObj.frame) {
```

Comment:
I see this checks more strictly (as opposed to previous version) that the subscription ignored is the exact object + frame that sent it – is that fixing an existing bug or required for the design change. Just wrapping my head around how I should write socket-handling code in the future.

Reply:
This is another one that I'm not too sure about. I believe this is the source of the `frames` typo below becoming a bug